glommio/src/io/bulk_io.rs

```diff
@@ -186,7 +186,7 @@ impl<V: IoVec> IOVecMerger<V> {
     }
 
     pub(super) fn flush(&mut self) -> Option<MergedIOVecs<V>> {
-        self.current.map(|x| MergedIOVecs {
+        self.current.take().map(|x| MergedIOVecs {
```

Comment:
Can you explain the context of this? What was the bug, what were you observing, and what's the expected behavior? I wrote this code a long time ago, and I admit I forgot most of the details 😅

Reply:
Sure. `MergedIOVecs` is used to merge (coalesce) I/Os inside `CoalescedReads` as its `merger`. `CoalescedReads` may [flush its internal merger](https://github.com/DataDog/glommio/blob/6d4aa8fb00b90fa9b6faa31d71a6d054ad043908/glommio/src/io/bulk_io.rs#L254) upon Pending and keep using the same `merger` afterwards. However, the flush method didn't reset the current I/O, and this means the current I/O will still be considered after it is flushed.

The added unit test demonstrates the problematic usage. Without this fix, the test failed. And this fix let the test pass.

Reply:
That makes sense to me, thanks!